packages/core/src/shaderlib/extra/particle.vs.glsl

```diff
@@ -40,25 +39,15 @@ void main() {
   v_color = a_color;
   v_uv = a_uv.xy;
   
-  // Real life time
-  float life = a_lifeAndSize.y + a_lifeAndSize.x;
+  // life time
+  float life = a_lifeAndSize.y;
+  float startTime = a_lifeAndSize.x;
 
   // Elapsed time
-  float deltaTime = max(mod(u_time, life) - a_lifeAndSize.x, 0.0);
+  float deltaTime = max(mod(u_time - startTime, life), 0.0);
 
-  bool isDying = false;
-
-  if (u_once || !u_active) {
-    isDying = true;
-  }
-
-  if ((isDying && u_time > life)) {
-    deltaTime = life;
-  }
-
-  // Not born means death, otherwise it will be displayed if not born
-  if (deltaTime == 0.0) {
-    deltaTime = life;
+  if ((u_once && u_time > life + startTime)) {
+    deltaTime = 0.0;
   }
 
   v_lifeLeft = 1.0 - deltaTime / a_lifeAndSize.y;
```

Comment:
v_lifeLeft = 1.0 - deltaTime / life is better

Reply:
fixed.